text/0308-cli-advisories.md

```diff
@@ -0,0 +1,332 @@
+| Affected component     | The CLI or the construct library (or both)                                                                                                                 | A non-empty subset of {CLI, Framework} | Yes        |
+| Affected version range | Version range using the semver format                                                                                                                      | semver                                 | No         |
+| Workaround             | Actions the user can take to mitigate the problem. | Free-form text                         | No         |
+
+In order to keep the issue human readable as well as easily parseable, each of
+these fields should be indicated by a Markdown header (`### Overview`, `###
+Affected component` etc). The piece of text following the header will be
+interpreted as the content of the field. See the template in the **Working
+backwards** section.
+
+#### GitHub workflow
+
+To be considered an advisory, a GitHub issue must be pinned and have the
+following tags: `p0` and a new tag, `advisory`. The `advisory` tag will work as
+an acknowledgement that author of the issue is making a conscious decision to
+publish an advisory that will be seen by potentially hundreds of thousands of
+users. Adding this tag will trigger a new workflow, which will inspect the issue
+and check whether its body contains the right data format (see **Data format**
+section). If the issue doesn’t comply with the specification, the workflow will
+post a message (similar to a CI report, for example).
+
+#### CLI logic
+
+The CLI will make unauthenticated requests to the GitHub Issues API and query
+for issues that match the tag system described in the **GitHub workflow**
+section. Then it will apply additional filters on the result:
+
+* Format filter: whether the body of the issue complies with the data format
+  specification.
+* Version filter: whether the version range contained in the issue matches the
+  CLI version or the framework version (depending on the affected component,
+  also present in the issue). To check which version of the framework is being
+  used, the CLI will read the cloud assembly (`tree.json` file, specifically).
+
+Issues that pass these filters will be displayed on the standard output.
+
+The GitHub API imposes a rate limit of 60 requests per hour for unauthenticated
+requests. To avoid hitting that limit, the CLI will cache the results for a set
+period of time (provisionally defined as 1 hour). The filtered advisories and
+the expiration time will be saved to a file in the `$HOME/.cdk/cache` folder.
+When the expiration time is reached, the cache is considered invalid and, at the
+next command execution, the CLI will make a new request to get fresh results.
```

Comment:
Perhaps we could add a sentence or two describing what the expected behavior should be if the advisories can't be retrieved (either due to network reasons, or due to GitHub API rate limits).